patterns-ts-cli/src/svg.ts

```diff
@@ -0,0 +1,71 @@
+/**
+ * Implementation of the svg command.
+ * 
+ * Read a pattern from a json file and write it to an svg file.
+ */
+
+import { readFile, writeFile } from "node:fs/promises";
+import path from "node:path";
+import { Pattern } from "@patternsoflife/patterns";
+import { PatternDeserializer } from "@patternsoflife/patterns/io";
+import { pattern2svg } from "@patternsoflife/patterns/visual";
+
+
+/**
+ * Read a json file and parse it.
+ */
+const readData = async (dataPath: string) => {
+  try {
+    const data = await readFile(dataPath, { encoding: 'utf8' });
+    return JSON.parse(data);
+  } catch (error) {
+    if (error instanceof Error) {
+      throw new Error(`Cannot read data from ${dataPath}:\n${error.message}`);
+    } else {
+      throw error;
+    }
+  }
+};
+
+
+/**
+ * Read a pattern from a json file and deserialize it.
+ */
+const readPattern = async (patternPath: string): Promise<Pattern> => {
+  const [
+    contextCatData, roleCatData, nodeDefData, patternData
+  ] = await Promise.all([
+    "definitions/context_categories.json",
+    "definitions/role_categories.json",
+    "definitions/node_definitions.json",
+    patternPath
+  ].map(readData));
+
+  const deserializer = new PatternDeserializer(
+    contextCatData, roleCatData, nodeDefData
+  );
+
+  return deserializer.deserializePattern(patternData);
+};
+
+
+/**
+ * Write a pattern from a json file to an svg file.
+ * 
+ * If the source file is pattern.json the output will be pattern.svg.
+ */
+export const writeSVG = async (patternPath: string) => {
+  try {
+    const destPath = `${path.parse(patternPath).name}.svg`;
+    const pattern = await readPattern(patternPath);
+    const svg = pattern2svg(pattern, { useDropShadows: true });
+    await writeFile(destPath, svg);
+  } catch (error) {
+    if (error instanceof Error) {
+      console.error(error.message);
```

Comment:
Maybe we can print the original stack here for easy debugging?

```suggestion
      console.error(error.stack ?? error.message);
```

Reply:
My thought here was we don't want to dump an entire stack trace if Jim simply made a typo in a filename. I could add a debug option though?

Reply:
No, I think that makes sense. Maybe we can add the debug if/when we realize we actually need it.

Reply:
@jessicabran I pushed a commit that prints the stack if the command is run with `--debug`